pom.xml

```diff
@@ -29,6 +29,7 @@
         <module>nifi-framework-api</module>
         <module>nifi-server-api</module>
         <module>nifi-bootstrap</module>
+        <module>nifi-code-coverage</module>
```

Comment:
I think the other possibie option, to avoid the new `nifi-code-coverage` sub-module, which may become out of step when new things are added to the repo, would be to add:

```xml
<properties>
  <jacoco.skip>true</jacoco.skip>
</properties>
```

to all sub-modules where we don't want to collect coverage, e.g. NAR and Bundle sub-modules. So this would remove the need for a new sub-module to maintain, but would mean updating lots of other existing sub-modules instead, then people understanding when to include the `skip` property for new sub-modules in future.

I don't think either option is ideal, but thought I'd raise the alternative for consideration - I'd be happy with either considered option

Reply:
Thanks for the feedback. I considered the option of individually running the JaCoCo [report](https://www.jacoco.org/jacoco/trunk/doc/report-mojo.html) goal, but I went with the [report-aggregate](https://www.jacoco.org/jacoco/trunk/doc/report-aggregate-mojo.html) goal and aggregating module to avoid changing individual Maven configurations.

I agree that neither solution is perfect, as each approach requires an extra element to consider when adding new modules. However, after making recent changes to remove the need for JavaDoc and source plugin skipping in individual modules, I went with the centralized approach. Another positive of this option is that if we don't see coverage information for a particular module, there is only one place to check, versus going through multiple modules.